sol/main.c

```diff
@@ -504,14 +541,44 @@ static int
 sol_stage1(void *arg)
 {
 	struct sol_config *sol_conf = arg;
-	int ret = get_queue_id(&sol_conf->net->back, QUEUE_TYPE_TX,
-		sol_conf->lcore_id, NULL);
-	if (ret < 0) {
-		SOL_LOG(ERR, "Cannot assign a TX queue for the back interface for lcore %u\n",
-			sol_conf->lcore_id);
+	int ret, i;
+
+	sol_conf->instances = rte_calloc(__func__, sol_conf->num_lcores,
```

Comment:
Use `rte_calloc_socket()` to be aligned with pull request #383.

I've looked for other cases to comment on, but I may have missed some spots, so please also review this patch for this kind of item.

Reply:
Thanks @AltraMayor ! In pull request #383 , we are assuming that all the GK blocks will be allocated on the same NUMA node. However, as we are now spreading the SOL blocks among the NUMA nodes, ideally, all the instances should be allocated according to its ``socket_id``.

Reply:
Pull request #383 is going to be merged before this pull request, so we must comply with pull request #383 here.

Notice that although the master branch is yet allocating all GK blocks on the same NUMA node, the code in pull request #383 doesn't require this assumption. The case in this comment thread is a problem that the GK blocks had to deal with as well, namely, the field `->instances` belongs to all instances. The code for GK block compromises and allocates the memory of this array in the same NUMA node of the first instance of the GK block. We can adopt this solution here as well.